lib/src/run.dart

```diff
@@ -144,3 +152,56 @@ stderr:
 
 $stderr""";
 }
+
+
+/// Arguments passed to [Process.run] .
+/// See [Process.run] for more details.
+class RunOptions {
+  final String workingDirectory;
+  final Map<String, String> environment;
+  final bool includeParentEnvironment;
+  final bool runInShell;
+  final Encoding stdoutEncoding;
+  final Encoding stderrEncoding;
+
+  RunOptions({this.workingDirectory, this.environment,
+      this.includeParentEnvironment: true, this.runInShell: false,
+      this.stdoutEncoding: SYSTEM_ENCODING,
+      this.stderrEncoding: SYSTEM_ENCODING});
+
+  /// Create a clone when it's necessary to modify the passed runOptions to
+  /// avoid modifying the argument.
+  RunOptions clone() {
+    return new RunOptions(
+        workingDirectory: workingDirectory,
+        environment: environment == null ? null : new Map.from(environment),
+        includeParentEnvironment: includeParentEnvironment,
+        runInShell: runInShell,
+        stdoutEncoding: stdoutEncoding,
+        stderrEncoding: stderrEncoding);
+  }
+}
+
+/// Arguments passed to [Process.start] .
+/// See [Process.start] for more details.
+class RunAsyncOptions {
```

Comment:
I don't think `RunAsyncOptions` is needed.

Firstly, `runAsync` is just like `run`, except for the output is live streamed to grinder's logging as the process is running.  So `stdoutEncoding` and `stderrEncoding` would be useful to `runAsync` (and its callers) as well to decode output before logging.

Secondly, the functionality of `runAsync` to throw an exception on non-zero exit code would not be available anyway when starting a process in `DETACHED` or `DETACHED_WITH_STDIO` mode.  And the functionality to automatically `log` the output would not be available in `DETACHED` mode.  So I don't think exposing the `mode` will be very useful.

So combining those two, both `run` and `runAsync` (and their callers) would benefit from exactly the fields you currently have defined in `RunOptions`.

Agree?